crates/eth-rpc/src/servers/net_rpc.rs

```diff
@@ -30,4 +33,12 @@ impl NetApiServer for NetRpc {
         // Kakarot RPC currently does not support peer-to-peer connections
         Ok(false)
     }
+
+    fn health(&self) -> Result<bool, io::Error> {
+        // Calls `starknet_blockNumber` method to check if it resolves
+        match self.kakarot_client.starknet_blockNumber() {
```

Comment:
I think this method does not exist, maybe you can check this by adding a health call at the beginning on integration.rs

I think the correct impl is:
`self.kakarot_client.starknet_provider().block_number()`

Reply:
Moreover, you'll need to handle the possible error, maybe with map_err for now, 
in the following way:
`self.kakarot_client.starknet_provider().block_number().await.map_err(EthApiError::from)?;`

Reply:
@Eikix  i just updated the PR. Again, not sure about my code.
Is it normal that i need to add `--no-verify` when i commit with git, otherwise i have issues committing. Indeed, with my rust extension, i can see some lines of code in core crate are not ok.

Reply:
I checked integration.rs file but i'm not sure about what i should do in there exactly. Could you give me some, more tips ?

Reply:
> @Eikix i just updated the PR. Again, not sure about my code. Is it normal that i need to add `--no-verify` when i commit with git, otherwise i have issues committing. Indeed, with my rust extension, i can see some lines of code in core crate are not ok. (more precisely, mod.rs and transaction.rs)

I think it is the precommit hook that returns an error, saying some of your code is not respecting the linter / compilation.

Reply:
> I checked integration.rs file but i'm not sure about what i should do in there exactly. Could you give me some, more tips ?

I think you'll need to slightly modify KakarotRpcModuleBuilder::new to have NetApi take a reference to KakarotClient, akin to other RPC modules

Then at the beginning of integration.rs, right before `let wallet= ...`
You can add a reqwest post request to net_health endpoint, 
endpoint address: `format!("http://localhost:{}", server_addr.port())`
endpoint selector: `net_health`
http word: `post`

take reference to https://docs.rs/reqwest/latest/reqwest/#json

Reply:
you should:
- rebase on main 
- run `make setup`, note that you may need to copy .env.example into your .env 
- then run test

Reply:
everything works perfectly now, thanks!

Reply:
@Eikix  i was trying to push my new commit but i'm having an issue with the pre hook i activated again : 

```
tristan@Tristans-MBP kakarot-rpc % git commit -m healthCheck
+cargo clippy --all -- -D warnings
    Checking kakarot-rpc-core v0.1.0 (/Users/tristan/Desktop/Dev/kakarot-rpc/crates/core)
error[E0308]: mismatched types
   --> crates/core/src/client/mod.rs:226:48
    |
226 |         let storage = contract_account.storage(&Uint256 { low: key_low.into(), high: key_high.into() }).call().await?;
    |                                        ------- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `&FieldElement`, found `&Uint256`
    |                                        |
    |                                        arguments to this method are incorrect
    |
    = note: expected reference `&starknet_crypto::FieldElement`
               found reference `&client::Uint256`
note: method defined here
   --> crates/core/src/client/mod.rs:51:1
    |
51  | abigen_legacy!(ContractAccount, "./artifacts/contract_account.json");
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = note: this error originates in the macro `abigen_legacy` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0308`.
error: could not compile `kakarot-rpc-core` (lib) due to previous error

```

This is not related to my modifications. This issue is in the core crate, mod.rs file. What am i supposed to do?

Reply:
Weird! Did you rebase on main?

Reply:
yes i did

Reply:
The codebase of kakarot RPC is the same and there is this mismatch of data type. Does this mean the codebase of main has an inconstancy?  how could this be possible with the pre commit hook ?

Reply:
Yes I agree it's weird! It shouldn't be the case, let me check

Reply:
The code in Kakarot RPC is up to date, did you try to pull main and then rebase ?

Reply:
i think i'm good now ! thank you